geode-core/src/main/java/org/apache/geode/internal/cache/versions/RegionVersionHolder.java

```diff
@@ -384,7 +370,11 @@ private void recordVersionWithBitSet(long version) {
   }
 
   private void setVersionInBitSet(long version) {
-    this.bitSet.set((int) (version - this.bitSetVersion));
+    long bitToSet = version - this.bitSetVersion;
+    if (bitToSet > BIT_SET_WIDTH) {
+      Assert.fail("Trying to set a bit larger than the size of the bitset " + bitToSet);
```

Comment:
Instead of Assert should we use `Math.toIntExact()`?

Reply:
This assertion is a little stronger than toIntExact. We are asserting that the bitToSet is less than or equal to BIT_SET_WIDTH (around 1K).

Reply:
Good point.  I'm ambivalent about using `Assert` instead of throwing an exception always.